packages/firestore/src/api/database.ts

```diff
@@ -2375,12 +2381,10 @@ export class CollectionReference<T = DocumentData>
 
   add(value: T): Promise<PublicDocumentReference<T>> {
     validateExactNumberOfArgs('CollectionReference.add', arguments, 1);
-    const convertedValue = this._converter
-      ? this._converter.toFirestore(value)
-      : value;
-    validateArgType('CollectionReference.add', 'object', 1, convertedValue);
     const docRef = this.doc();
-    return docRef.set(value).then(() => docRef);
+    return ((docRef as unknown) as DocumentReference)
+      .set(value, undefined, 'CollectionReference.add')
+      .then(() => docRef);
```

Comment:
Couldn't you just leave the original code as is and do:

return docRef.withConverter(defaultConverter).set(value).then(() => docRef)

?